test/socket.jl

```diff
@@ -174,3 +174,56 @@ begin
     send(b, ip"::1", port, "Hello World")
     wait(tsk)
 end
+
+begin
+    default_port = UInt16(11011)
+    default_addr = IPv4("127.0.1.1")
+
+    sock = Base.TCPServer()
+    bind(sock,Base.InetAddr(default_addr,default_port))
+    listen(sock)
+
+    new_addr, new_port = getsockname(sock)
+
+    @test default_addr == new_addr
+    @test default_port == new_port
+    close(sock)
+end
+
+@unix_only begin
```

Comment:
see #9882 and #10335

On Thu, Apr 30, 2015 at 2:17 PM, Art <notifications@github.com> wrote:

> In test/socket.jl
> <https://github.com/JuliaLang/julia/pull/11012#discussion_r29455402>:
>
> > +begin
> > +    default_port = UInt16(11011)
> > +    default_addr = IPv4("127.0.1.1")
> > +
> > +    sock = Base.TCPServer()
> > +    bind(sock,Base.InetAddr(default_addr,default_port))
> > +    listen(sock)
> > +
> > +    new_addr, new_port = getsockname(sock)
> > +
> > +    @test default_addr == new_addr
> > +    @test default_port == new_port
> > +    close(sock)
> > +end
> > +
> > +@unix_only begin
>
> Appveyor failed of this test on Windows, I guess they do not have IPv6
> available on test machine. I'll turn if back. Problem is how to tell if
> IPv6 is on or not before executing this test. Or should I remove it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/11012/files#r29455402>.
>